walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -52,50 +55,60 @@
-            optionalAttendeeAvailableTimes.get(optionalAttendee).addAll(availableTimes);
+        HashMap<String, Integer> optionalAttendeesAvailableTimeValues = new HashMap<String, Integer>();
+        for (int i = 0; i < optionalAttendeesAvailableTimes.size(); i++) {
+            optionalAttendeesAvailableTimeValues.put(i + "0", optionalAttendeesAvailableTimes.get(i).start());
+            optionalAttendeesAvailableTimeValues.put(i + "1", optionalAttendeesAvailableTimes.get(i).end());
         }
-        HashMap<TimeRange, Integer> optimalAvailableTimes = new HashMap<TimeRange, Integer>();
-        for (Map.Entry<String, ArrayList<TimeRange>> entry : optionalAttendeeAvailableTimes.entrySet()) {
-            ArrayList<TimeRange> availableTimes = entry.getValue();
-            for (TimeRange availableTime : availableTimes) {
-                for (TimeRange mandatoryAttendeesAvailableTime : mandatoryAttendeesAvailableTimes) {
-                    if ((mandatoryAttendeesAvailableTime.contains(availableTime) == true) || (availableTime.contains(mandatoryAttendeesAvailableTime) == true)) {
-                        if (optimalAvailableTimes.containsKey(mandatoryAttendeesAvailableTime)) {
-                            int count = optimalAvailableTimes.get(mandatoryAttendeesAvailableTime);
-                            optimalAvailableTimes.put(mandatoryAttendeesAvailableTime, count + 1);
-                        } else {
-                            optimalAvailableTimes.put(mandatoryAttendeesAvailableTime, 1);
-                        }
-                    }
-                }
+        // Sort time values.
+        ArrayList<Map.Entry<String, Integer>> list = new ArrayList<Map.Entry<String, Integer>>();
+        Collections.sort(list, new Comparator<Map.Entry<String, Integer>>() {
+            public int compare(Map.Entry<String, Integer> o1, Map.Entry<String, Integer> o2) {
+                return (o1.getValue()).compareTo(o2.getValue());
             }
+        });
+        HashMap<String, Integer> optionalAttendeesAvailableTimeValuesSorted = new LinkedHashMap<String, Integer>();
+        for (Map.Entry<String, Integer> entry : list) {
+            optionalAttendeesAvailableTimeValuesSorted.put(entry.getKey(), entry.getValue());
         }
+        // Find the time values when the maximum number of attendees are available.
         int maximum = 0;
-        ArrayList<TimeRange> optimalAvailableTime = new ArrayList<TimeRange>();
-        for (Map.Entry<TimeRange, Integer> entry : optimalAvailableTimes.entrySet()) {
-            TimeRange availableTime = entry.getKey();
-            int count = entry.getValue();
+        ArrayList<Integer> optionalAttendeesOptimalAvailableTimes = new ArrayList<Integer>();
+        int count = 0;
+        for (Map.Entry<String, Integer> entry : optionalAttendeesAvailableTimeValuesSorted.entrySet()) {
+            switch (entry.getKey().charAt(1)) {
```

Comment:
will .charAt(1) work if the number of optional attendees is more than 9? in the for loop on line 59, you add i + "0" or "1". if i is 12, that would translate to 120, and charAt(1) would return 2. you probably need to calculate the length of the entry.getKey(). however, I wonder if there's another way you can store the start/end information. 

also, i thought switch cases were supposed to have a default case..is that not the case in java?

Reply:
I handled the case by changing the order of the index and the start or end information.  Should I make the code more readable by using “S” and “E” to store the start or end information?

Reply:
The default case is optional in Java.